src/engine/ContextDependentPiecewiseLinearConstraint.cpp

```diff
@@ -0,0 +1,189 @@
+/*********************                                                        */
+/*! \file ContextDependentPiecewiseLinearConstraint.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aleksandar Zeljic
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** A context dependent implementation of ContextDependentPiecewiseLinearConstraint class
+ **/
+
+#include "ContextDependentPiecewiseLinearConstraint.h"
+
+#include "Statistics.h"
+
+ContextDependentPiecewiseLinearConstraint::ContextDependentPiecewiseLinearConstraint()
+   : _numCases( 0 )
+   , _boundManager( nullptr )
+   , _context( nullptr )
+   , _cdConstraintActive( nullptr )
+   , _cdPhaseStatus( nullptr )
+   , _cdInfeasibleCases( nullptr )
+{
+}
+
+ContextDependentPiecewiseLinearConstraint::ContextDependentPiecewiseLinearConstraint(
+    unsigned numCases )
+   : _numCases( numCases )
+   , _boundManager( nullptr )
+   , _context( nullptr )
+   , _cdConstraintActive( nullptr )
+   , _cdPhaseStatus( nullptr )
+   , _cdInfeasibleCases( nullptr )
+{
+}
+
+void ContextDependentPiecewiseLinearConstraint::setActiveConstraint( bool active )
+{
+    if ( nullptr != _cdConstraintActive )
+        *_cdConstraintActive = active;
+    else
+        _constraintActive = active;
+}
+
+bool ContextDependentPiecewiseLinearConstraint::isActive() const
+{
+    if ( nullptr != _cdConstraintActive )
+        return *_cdConstraintActive;
+    else
+        return _constraintActive;
+}
+
+void ContextDependentPiecewiseLinearConstraint::registerBoundManager(
+    BoundManager *boundManager )
+{
+    ASSERT( nullptr == _boundManager );
+    _boundManager = boundManager;
+}
+
+void ContextDependentPiecewiseLinearConstraint::initializeCDOs(
+    CVC4::context::Context *context )
+{
+    ASSERT( nullptr == _context );
+    _context = context;
+
+    initializeCDActiveStatus();
+    initializeCDPhaseStatus();
+    initializeCDInfeasibleCases();
+}
+
+void ContextDependentPiecewiseLinearConstraint::initializeCDInfeasibleCases()
+{
+    ASSERT( nullptr != _context );
+    ASSERT( nullptr == _cdInfeasibleCases );
+    _cdInfeasibleCases = new ( true ) CVC4::context::CDList<PhaseStatus>( _context );
+}
+
+void ContextDependentPiecewiseLinearConstraint::initializeCDActiveStatus()
+{
+    ASSERT( nullptr != _context );
+    ASSERT( nullptr == _cdConstraintActive );
+    _cdConstraintActive = new ( true ) CVC4::context::CDO<bool>( _context, true );
```

Comment:
shouldn't we initialize to _constraintActive?

Reply:
Fixed